	* gio/gsocket.c: Fix build on Solaris 8 by providing needed macros
	defined on Solaris 10.

--- gio/gsocket.c.old	2010-01-07 01:19:12.000000000 +0100
+++ gio/gsocket.c	2010-01-29 13:07:02.275043000 +0100
@@ -55,6 +55,16 @@
 
 #include "gioalias.h"
 
+#if defined (sun) && !defined (CMSG_SPACE)
+/* Amount of space + padding needed for a message of length l */
+#define CMSG_SPACE(l)                                                   \
+        ((unsigned int)_CMSG_HDR_ALIGN(sizeof (struct cmsghdr) + (l)))
+
+/* Value to be used in cmsg_len, does not include trailing padding */
+#define CMSG_LEN(l)                                                     \
+        ((unsigned int)_CMSG_DATA_ALIGN(sizeof (struct cmsghdr)) + (l))
+#endif
+
 /**
  * SECTION:gsocket
  * @short_description: Low-level socket object
